scripts/umbrel-os/external-storage/monitor

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+
+# Load monitor-lib
+check_dependencies () {
+  for cmd in "$@"; do
+    if ! command -v $cmd >/dev/null 2>&1; then
+      echo "This script requires \"${cmd}\" to be installed"
+      exit 1
+    fi
+  done
+}
+
+check_dependencies readlink dirname
+UMBREL_ROOT="$(dirname $(readlink -f "${BASH_SOURCE[0]}"))/../.."
+
+source "${UMBREL_ROOT}/scripts/monitor-lib"
+
+
+# Monitor external mount
+block_device="${1}"
+MOUNT_POINT="${2}"
+
+check_mount () {
+    # Check if Umbrel root is still bind mounted
+    if ! df -h "${UMBREL_ROOT}" | grep --quiet '/dev/sd'; then
+        echo "External device not bind mounted to Umbrel root"
+        return 1
+    fi
+
+    # Check if device is still mounted
+    if ! grep -q "${MOUNT_POINT}" /proc/mounts; then
+        echo "External device not mounted at ${MOUNT_POINT}"
+        return 1
+    fi
+
+    # Check if device is read-only
+    READ_ONLY=$(cat /sys/block/${block_device}/ro)
+    if [[ "${READ_ONLY}" == "1" ]]; then
+        echo "External block device '${block_device}' set to 'read-only'"
+        return 1
+    fi
+
+    return 0
+}
+
+run_monitor () {
+    while check_mount; do
+        sleep 1
+    done
+
+    # 'mount-fail' gets triggered if 'check_mount' fails
+    trigger="mount-fail"
+    touch "${UMBREL_ROOT}/events/signals/${trigger}"
+}
+
+echo "Starting external drive monitoring process..."
+run_monitor
```

Comment:
I wasn't sure is this should be run with an `&` at the end as well given that the parent script is called with one inside the `mount` script

Reply:
Nope, no need to eun as a background service, it's fine as is.